### DISCUSSION TITLE
Salesforce as Auth Provider (with benefits)

### DISCUSSION THREAD

**Discussion by multimanic** [supabase/supabase#11477] [Questions]
I'm trying to build an app that where users can log in using their Salesforce account and at the same time collect/save their OAuth access token and refresh token, so the app can query Salesforce for some particular data on its own.

I was hoping SB would also support Salesforce as an Auth provider - but no luck. Any ideas how to build this from scratch without loosing all the other benefits of the build in auth (RLI, auth.getUser(), etc.)?

Thanks

**Comment by multimanic** [comment]
I guess what I'm looking for is the equivalent of this: https://firebase.google.com/docs/auth/web/custom-auth
Sorry for using the f-word in this discussion :)

**Comment by ckaminski** [comment]
I'm thinking the same thing.  I think the simplest might be to take one of the existing auth providers in supabase/gotrue and developing it yourself.  I'm no familiar with Go so it's going to be slow going for me.  I'm prototyping with Azure, which is often an auth provider for SF logins anyway.
